### ISSUE TITLE
[kube-prometheus-stack] node selections has only None on dashboard Kubernetes / Compute Resources / Node (Pods)

### ISSUE THREAD

**Issue by wu105** [prometheus-community/helm-charts#1655]
### Describe the bug a clear and concise description of what the bug is.

On grafana dashboard `Kubernetes / Compute Resources / Node (Pods)`, the `node` drop down offers only `None`.  

### What's your helm version?

version.BuildInfo{Version:"v3.6.0", GitCommit:"7f2df6467771a75f5646b7f12afb408590ed1755", GitTreeState:"clean", GoVersion:"go1.16.3"}

### What's your kubectl version?

Client Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.2+vmware.1", GitCommit:"54e7e68e30dd3f9f7bb4f814c9d112f54f0fb273", GitTreeState:"clean", BuildDate:"2021-06-28T22:17:36Z", GoVersion:"go1.16.5", Compiler:"gc", Platform:"linux/amd64"} Server Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.2+vmware.1", GitCommit:"54e7e68e30dd3f9f7bb4f814c9d112f54f0fb273", GitTreeState:"clean", BuildDate:"2021-06-28T22:12:04Z", GoVersion:"go1.16.5", Compiler:"gc", Platform:"linux/amd64"}

### Which chart?

prometheus-community/kube-prometheus-stack from https://prometheus-community.github.io/helm-charts

### What's the chart version?

kube-prometheus-stack-27.0.1

### What happened?

Dashboard `General / Kubernetes / Compute Resources / Node (Pods)` node dropdown shows only `None` as a selection.

### What you expected to happen?

Dashboard `General / Kubernetes / Compute Resources / Node (Pods)` node dropdown shows all the node names as selection.

### How to reproduce it?

install Prometheus and check the dashboard.

### Enter the changed values of values.yaml?

Here is the values (verbatim except `something`):
```
helm get values prometheus -n monitoring
USER-SUPPLIED VALUES:
alertmanager:
  ingress:
    enabled: true
    hosts:
    - prometheus-alertmanager.z2wl.something.com
    pathType: ImplementationSpecific
    tls:
    - hosts:
      - prometheus-alertmanager.z2wl.something.com
      secretName: tls-prometheus
grafana:
  ingress:
    enabled: true
    hosts:
    - prometheus-grafana.z2wl.something.com
    pathType: Prefix
    tls:
    - hosts:
      - prometheus-grafana.z2wl.something.com
      secretName: tls-prometheus
kubeControllerManager:
  service:
    port: 10257
    targetPort: 10257
  serviceMonitor:
    https: true
    insecureSkipVerify: true
    serverName: localhost
kubeEtcd:
  serviceMonitor:
    caFile: /etc/prometheus/secrets/tls-etcdcerts/ca.crt
    certFile: /etc/prometheus/secrets/tls-etcdcerts/healthcheck-client.crt
    insecureSkipVerify: true
    keyFile: /etc/prometheus/secrets/tls-etcdcerts/healthcheck-client.key
    scheme: https
kubeScheduler:
  service:
    port: 10259
    targetPort: 10259
  serviceMonitor:
    https: true
    insecureSkipVerify: true
    serverName: localhost
prometheus:
  ingress:
    enabled: true
    hosts:
    - prometheus.z2wl.something.com
    pathType: ImplementationSpecific
    tls:
    - hosts:
      - prometheus.z2wl.something.com
      secretName: tls-prometheus
  prometheusSpec:
    secrets:
    - tls-etcdcerts
``` 

### Enter the command that you execute and failing/misfunctioning.

n/a

### Anything else we need to know?

Dashboard `General / Kubernetes / Compute Resources / Namespace (Workloads)` has similar selection for `namespace`.

The issue was not present in the middle of November, but showed up over the new year's weekend.  This reports on the latest chart version on kubernetes v1.21.2, but similar issue is observed using (very) older charts on k8s 1.12.

**Comment by jsalatiel** [comment]
The last working version for me is 23.3.2. After that all I get in grafana is no data.

**Comment by groundhog2k** [comment]
I can confirm, that something is broken since version 24.0.0 - see also: https://github.com/prometheus-community/helm-charts/issues/1661

**Comment by monotek** [comment]
Please read the upgrading section: 
https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/README.md#upgrading-an-existing-release-to-a-new-major-version

You also might want to set the release label in kube-state-metrics. See: 
https://github.com/prometheus-community/helm-charts/issues/1631

**Comment by wu105** [comment]
@monotek On reference to #1631 - as of kube-prometheus-stack-27.0.1, all the release labels are "prometheus", thus the work around in #1631 need to set the labels to "prometheus", and it resolved the issue for kube-prometheus-stack-27.0.1 -- Many thanks.

I am still investigating the issue on a older release - it was working a month earlier and no changes to the installation since as far as I am concerned, except some nodes are drained and the pods are moved.

**Comment by francardoso93** [comment]
`kube-state-metrics.releaseLabel=true` did not automatically add the release label to the serviceMonitor.
